vector/src/main/java/im/vector/app/features/home/room/detail/timeline/format/NoticeEventFormatter.kt

```diff
@@ -118,6 +121,19 @@ class NoticeEventFormatter @Inject constructor(
         }
     }
 
+    private fun formatPinnedEvent(event: Event, disambiguatedDisplayName: String): CharSequence? {
+        val idsOfPinnedEvents: MutableList<String> = event.getIdsOfPinnedEvents() ?: return null
+        val previousIdsOfPinnedEvents: MutableList<String>? = event.getPreviousIdsOfPinnedEvents()
+        // A message was pinned
+        val pinnedMessageString = if (event.resolvedPrevContent() == null || previousIdsOfPinnedEvents != null && previousIdsOfPinnedEvents.size < idsOfPinnedEvents.size) {
+            sp.getString(R.string.user_pinned_message, disambiguatedDisplayName)
+        // A message was unpinned
+        } else {
+            sp.getString(R.string.user_unpinned_message, disambiguatedDisplayName)
+        }
+        return pinnedMessageString
+    }
```

Comment:
Better to use `event.isSentByCurrentUser()` to have a nicer wording when the user is the sender of the state event.
Also for the string resource, you could use the existing naming convention. Start the resource name with `notice_`.